### PR TITLE
support explicit this references in natvis files

### PR DESCRIPTION
**Pull request by Trass3r**
See https://github.com/microsoft/MIEngine/issues/197#issuecomment-692208432.

### CODE REVIEW

File: src/MIDebugEngine/Natvis.Impl/Natvis.cs [hunk 1]
```diff
@@ -1040,12 +1040,19 @@ private string ReplaceNamesInExpression(string expression, IVariableInformation
...
-                        {
-                            return "(" + var.FullName() + ")";
-                        }
+                    {
+                        if (variable == null)
+                            return null;
+
+                        // replace explicit this references
+                        if (m.Value == "this")
+                            return "(&" + variable.FullName() + ")";
```
Comment (Trass3r) [inline src/MIDebugEngine/Natvis.Impl/Natvis.cs:1049]: Not sure if additional parentheses or special handling for corner cases like pointers might be necessary here.

**Comment by Trass3r** [comment]
@gregg-miskelly would you like to review?

**Comment by Trass3r** [comment]
@WardenGnaw ?

**Comment by Trass3r** [comment]
@WardenGnaw @gregg-miskelly

**Comment by Trass3r** [comment]
Workflow runs need to be approved.

**Comment by gregg-miskelly** [comment]
Sorry for the delay. WardenGnaw and I have both been on vacation.

**Review by gregg-miskelly — APPROVED** [review]

**Comment by WardenGnaw** [comment]
> The test failure seems unrelated to natvis. It's also lldb specific which is hard to test on Linux (I kept running into problems while trying).
> 
> ```
> Running command setVariable (myint=39+nonexistingint)
> Command 'setVariable' expecting response: CommandResponse (Fail)
> ERROR: Running command 'setVariable'. Exception thrown.
> DARException: Expected response not found before timeout.
> Expected =
> 1. Not Found: {"success":false,"command":"setVariable"}
> 
> Actual Responses =
> 
> DebugAdapterRunner.DebugAdapterCommand.Run(DebugAdapterRunner runner) in /Users/runner/work/MIEngine/MIEngine/test/DebugAdapterRunner/DebugAdapterCommand.cs:line 271
> DebuggerTesting.OpenDebug.Commands.Command`1.Run(DebugAdapterRunner darRunner, ILoggingComponent log, IEvent[] expectedEvents) in /Users/runner/work/MIEngine/MIEngine/test/DebuggerTesting/OpenDebug/Commands/Command.cs:line 164
> ```

Sorry, this is unfortunately a flaky test issue but we keep it active to track it in the nightly test runs. 

LGTM.

**Review by WardenGnaw — APPROVED** [review]

**Pull request merged**